### ISSUE TITLE
Support passing options to acm certificate requests

### ISSUE THREAD

**Issue by starpebble** [terraform-providers/terraform-provider-aws#6789]
### Community Note

* Please vote on this issue by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
* Please do not leave "+1" or "me too" comments, they generate extra noise for issue followers and do not help prioritize the request
* If you are interested in working on this issue or have submitted a pull request, please leave a comment

### Description

Security enhancement: support creating aws acm certificates with options.

### New or Affected Resource(s)

aws_acm_certificate

### Potential Terraform Configuration

```
resource "aws_acm_certificate" "cert" {
  domain_name       = "example.com"
  validation_method = "DNS"

  options {
    CertificateTransparencyLoggingPreference = "DISABLED"
  }

  tags {
    Environment = "test"
  }

  lifecycle {
    create_before_destroy = true
  }
}
```

### References

acm certificates may be created with options.  A command line example

```
aws acm request-certificate \
--domain-name www.example.com \
--validation-method DNS \
--options CertificateTransparencyLoggingPreference=DISABLED \
--idempotency-token 184627
```

To test the enhancement, create certificates with transparency logging turned off.  Disabling transparency logging is a security best practice.  

Ideally, any future options will be automatically supported by the aws terraform provider when aws adds them to the aws go sdk.  That said, supporting CertificateTransparencyLoggingPreference is a good start.

References:

[1] request-certificate CLI documentation - https://docs.aws.amazon.com/cli/latest/reference/acm/request-certificate.html

[2] Disabling transparency logging description https://docs.aws.amazon.com/acm/latest/userguide/acm-bestpractices.html#best-practices-transparency

[3] Alternate issue proposal: https://github.com/terraform-providers/terraform-provider-aws/issues/5696

* #0000

**Comment by bflad** [comment]
Support for the new `options` configuration block and its `certificate_transparency_logging_preference` argument has been merged and will release with version 2.23.0 of the Terraform AWS Provider, later this week. Thanks to @mattburgess for the implementation. 👍

**Comment by starpebble** [comment]
I agree, the lower case with underscore convention is probably best.  

This is an extra nice security feature. Probably should be the default but we can’t debate that one.
